### PR TITLE
Have array_to_json_array support MapArray

### LINKED ISSUE #4297: Have array_to_json_array support Map
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
I have data that is off the type `MapArray`, but `array_to_json_array` currently does not support it. Using it with `MapArray` data type leads to the error

```
Json error: data type Map(...not supported
```

**Describe the solution you'd like**
It will be nice if array_to_json_array supports `MapArray`.

**Additional context**
`arrow-json = "40.0.0"`

### PR DESCRIPTION
**Pull request by dadepo**
# Which issue does this PR close?

Closes #4297

# Rationale for this change
 
Extended `array_to_json_array` to support `MapArray`

### CODE REVIEW

File: arrow-array/src/array/map_array.rs [hunk 1]
```diff
@@ -284,6 +292,18 @@ impl Array for MapArray {
     }
 }
 
+impl<'a> ArrayAccessor for &'a MapArray {
+    type Item = ArrayRef;
```
Comment (dadepo) [inline arrow-array/src/array/map_array.rs:296]: @tustvold any thoughts on having the iterator return `ArrayRef` instead of `StructArray`?

Comment (tustvold) [reply arrow-array/src/array/map_array.rs:296]: `StructArray` should be strictly more useful?

Comment (dadepo) [reply arrow-array/src/array/map_array.rs:296]: Yeah, just thinking without that, the consumer will always be forced to cast back to `StructArray`. 

I'll update it in a bit

**Review by tustvold — APPROVED** [review]

File: arrow-json/src/writer.rs [hunk 2]
```diff
@@ -202,6 +203,15 @@ pub fn array_to_json_array(array: &ArrayRef) -> Result<Vec<Value>, ArrowError> {
             let jsonmaps = struct_array_to_jsonmap_array(array.as_struct())?;
             Ok(jsonmaps.into_iter().map(Value::Object).collect())
         }
+        DataType::Map(_, _) => as_map_array(array)
+            .iter()
+            .map(|maybe_value| match maybe_value {
+                Some(v) => Ok(Value::Array(array_to_json_array(
```
Comment (tustvold) [inline arrow-json/src/writer.rs:209]: We should probably change array_to_json_array to take `&dyn Array` to avoid the needless Arc-wrapping

**Pull request merged**